syntax = "proto3";

package zmq_message;

enum MessageType {
  MESSAGE_OHLCData = 0;
}

message OHLCData {
    int64 timestamp = 1;
    uint32 tickerId = 2;
    double open = 3;
    double high = 4;
    double low = 5;
    double close = 6;
    uint64 volume = 7;
    int64 interval = 8;
}

message Wrapper {
  MessageType type = 1;
  OHLCData data = 2;  // Directly include OHLCData as a field
}
